shim/batch.go

```diff
@@ -0,0 +1,96 @@
+// Copyright the Hyperledger Fabric contributors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package shim
+
+import (
+	"github.com/hyperledger/fabric-protos-go-apiv2/peer"
+)
+
+type batchRecordType int
+
+const (
+	dataKeyType batchRecordType = iota
+	metadataKeyType
+)
+
+type batchKey struct {
+	Collection string
+	Key        string
+	Type       batchRecordType
+}
+
+type writeBatch struct {
+	writes map[batchKey]*peer.WriteRecord
+}
+
+func newWriteBatch() *writeBatch {
+	return &writeBatch{
+		writes: make(map[batchKey]*peer.WriteRecord),
+	}
+}
+
+func (b *writeBatch) Writes() []*peer.WriteRecord {
+	if b == nil {
+		return nil
+	}
+
+	var results []*peer.WriteRecord
```

Comment:
Save on copying a slice when adding elements

I suggest replacing the line 
`var results []*peer.WriteRecord`
to 
`results := make([]*peer.WriteRecord, 0, len(b.writes))`

Reply:
Your code didn’t do that but I am happy to make the change as soon as I get a chance.